handshake/src/prelude.rs

```diff
@@ -0,0 +1,246 @@
+            pub fn null() -> Self {
+                Self(vec![])
+            }
+
+            /// Return a reference to the underlying bytes
+            pub fn items(&self) -> &[u8] {
+                &self.0
+            }
+
+            /// Set the underlying items vector.
+            pub fn set_items(&mut self, v: Vec<u8>) {
+                self.0 = v
+            }
+
+            /// Push an item to the item vector.
+            pub fn push(&mut self, i: u8) {
+                self.0.push(i)
+            }
+
+            /// Return the length of the item vector.
+            pub fn len(&self) -> usize {
+                self.0.len()
+            }
+
+            /// Return true if the length of the item vector is 0.
+            pub fn is_empty(&self) -> bool {
+                self.len() == 0
+            }
+
+            /// Determine the byte-length of the vector length prefix
+            pub fn len_prefix(&self) -> u8 {
+                coins_core::ser::prefix_byte_len(self.len() as u64)
+            }
+
+            /// Insert an item at the specified index.
+            pub fn insert(&mut self, index: usize, i: u8) {
+                self.0.insert(index, i)
+            }
+        }
+
+        impl From<&[u8]> for $wrapper_name {
+            fn from(v: &[u8]) -> Self {
+                Self(v.to_vec())
+            }
+        }
+
+        impl From<Vec<u8>> for $wrapper_name {
+            fn from(v: Vec<u8>) -> Self {
+                Self(v)
+            }
+        }
+
+        impl std::ops::Index<usize> for $wrapper_name {
+            type Output = u8;
+
+            fn index(&self, index: usize) -> &Self::Output {
+                &self.0[index]
+            }
+        }
+
+        impl std::ops::Index<std::ops::Range<usize>> for $wrapper_name {
+            type Output = [u8];
+
+            fn index(&self, range: std::ops::Range<usize>) -> &[u8] {
+                &self.0[range]
+            }
+        }
+
+        impl std::ops::IndexMut<usize> for $wrapper_name {
+            fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+                &mut self.0[index]
+            }
+        }
+
+        impl std::iter::Extend<u8> for $wrapper_name {
+            fn extend<I: std::iter::IntoIterator<Item=u8>>(&mut self, iter: I) {
+                self.0.extend(iter)
+            }
+        }
+
+        impl std::iter::IntoIterator for $wrapper_name {
+            type Item = u8;
+            type IntoIter = std::vec::IntoIter<u8>;
+
+            fn into_iter(self) -> Self::IntoIter {
+                self.0.into_iter()
+            }
+        }
+    }
+}
+
+/// Make a new marked hash of length 32.
+#[macro_export]
+macro_rules! mark_hash256 {
+    (
+        $(#[$outer:meta])*
+        $hash_name:ident
+    ) => {
+        $(#[$outer])*
+        #[derive(Hash, serde::Serialize, serde::Deserialize, Copy, Clone, Default, Debug, Eq, PartialEq, PartialOrd, Ord)]
+        pub struct $hash_name(pub Blake2b256Digest);
```

Comment:
This macro needs to also be generic over the digest, in `bitcoins` its `Hash256Digest` and here it is `Blake2b256Digest`